### PR TITLE
miscellaneous cleanups

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

A couple of minor cleanups I haven't bothered to push upstream yet.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
going to split the ch4/am request changes to another PR.
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/whitespace

**Comment by raffenet** [comment]
test:mpich/whitesapce

File: src/mpid/ch4/include/mpidpre.h [hunk 1]
```diff
@@ -73,15 +73,6 @@ typedef enum {
...
 
-typedef struct MPIDIG_sreq_t {
-    /* persistent send fields */
-    const void *src_buf;
-    MPI_Aint count;
-    MPI_Datatype datatype;
-    int rank;
-    MPIR_Context_id_t context_id;
-} MPIDIG_sreq_t;
-
```
[the remark this answers is not in the document]
Comment (raffenet) [reply src/mpid/ch4/include/mpidpre.h]: These changes are dropped from this PR.

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**